### ISSUE TITLE
`deno types` to output typings to a single file

### ISSUE THREAD

**Issue by ebebbington** [denoland/deno#9377]
**What**

As it stands, there doesn't seem to be a way to generate typings for a file, or files, using Deno. It *might* be possible with `Deno.emit`, using the 2nd argument (`CompilerOptions`), but i haven't checked this and I'm not too confident it can.

It also seems like `deno types` could be expanded upon, ie there's a great opportunity to expand upon this subcommand.

But what I propose is for `deno types` (and ideally a programatic way **as well**) to create the typings for the given file or directory **that outputs to a single file**

**How**

```
$ deno types src/mod.ts >> dist/mod.d.ts
```

```ts
await Deno.emit("./src/mod.ts", { typings: true })
```

**Why**

Well say I have a project that uses typescript. I plan to move over from Node to Deno, but I need typings. It seems impossible to do this with Deno, halting on that projects progress to make the beautiful journey of moving to Deno. This issue. is specifically with emitting a bundled `.d.ts` file.

Whilst I can use `Deno.emit(file, { compilerOptions: { declaration: true }})`, it still doesn't provide the bundled typings

**Example**

Here is an example, of me wanting to create a single bundled `.d.ts` file using the `emit` method, but it not actually providing typings:

```ts
// src/mod.ts
export * from "./something.ts"
export * from "./test.ts"
```

```ts
// compile.ts
const stuff = await Deno.emit("./src/mod.ts", {
  compilerOptions: {
    "declaration": true,
    "emitDeclarationOnly": true,
  }
})
const keys = Object.keys(stuff.files)
const key = keys.find(k => k.includes("mod.ts.d.ts")) as string
console.log(stuff.files[key])
```

```
/// <amd-module name="file:///.../.../tmp/src/mod.ts" />
export * from "./something.ts";
export * from "./test.ts";
```

As you can see, it doesn't provide any typings

**Comment by 0kku** [comment]
Specifically, interoperability between Deno and TSC seems unachievable at present. Deno doesn't understand TSC's imports, nor TSC's .d.ts files because those also use the incorrect file extensions. On the other hand, TSC doesn't understand Deno's imports. There doesn't even seem to be any tools out there that would allow automating converting between the two formats. Ideally, you'd be able to bundle the typings of a project into a single .d.ts file in order to avoid the importing conflicts and allow you to be able to use the same code base for both Deno and TSC consumers.

**Comment by kitsonk** [comment]
Duplicate of #3385 

(You can already set `declaration: true` in the compiler options in `Deno.emit()` for single file declarations, just like you would with `tsc`)

**Comment by jespertheend** [comment]
Does `declaration: true` still work? It doesn't seem to do anything, and it appears in a list of ignored options:
https://github.com/denoland/deno/blob/1fb5858009f598ce3f917f9f49c466db81f4d9b0/cli/config_file.rs#L86-L90

**Comment by jespertheend** [comment]
Oh never mind, it seems `declaration: true` does in fact do something. But if you try to emit the same file a second time, it keeps emitting `d.ts` files regardless of your `declaration` setting. Only after you clear the Deno cache will it stop emitting them.
